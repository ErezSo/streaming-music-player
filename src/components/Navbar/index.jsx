import React from "react";
import { Link } from "react-router-dom";
import ReactAudioPlayer from "react-audio-player";
import { MusicPlayer } from "../index";
const styles = {
  header: {
    display: "inline-block"
  },
  link: {
    padding: "22px 34px"
  },
  text: {
    paddingRight: "2em",
    float: "right"
  }
};

function Navbar({ song = "" }) {
  return (
    <nav className="navbar navbar-default">
      <div className="container-fluid">
        <div className="navbar-header" style={styles.header}>
          <Link className="navbar-brand" style={styles.link} to="/">
            Home
          </Link>
        </div>
        <MusicPlayer song={song} />
      </div>
    </nav>
  );
}

export default Navbar;
