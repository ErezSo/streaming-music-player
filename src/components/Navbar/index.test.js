import React from "react";
import { shallow } from "enzyme";
import Navbar from "./";

describe("Navbar", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Navbar />);
  });

  it("renders", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("renders correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
