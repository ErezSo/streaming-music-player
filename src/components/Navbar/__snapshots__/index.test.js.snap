// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
"<nav className=\\"navbar navbar-default\\">
  <div className=\\"container-fluid\\">
    <div className=\\"navbar-header\\" style={{...}}>
      <Link className=\\"navbar-brand\\" style={{...}} to=\\"/\\" replace={false}>
        Home
      </Link>
    </div>
    <div className=\\"navbar-text navbar-right\\" style={{...}}>
      <t src=\\"\\" autoPlay={true} controls={true} className=\\"\\" controlsList=\\"\\" crossOrigin=\\"\\" id=\\"\\" listenInterval={10000} loop={false} muted={false} onAbort={[Function: onAbort]} onCanPlay={[Function: onCanPlay]} onCanPlayThrough={[Function: onCanPlayThrough]} onEnded={[Function: onEnded]} onError={[Function: onError]} onListen={[Function: onListen]} onPause={[Function: onPause]} onPlay={[Function: onPlay]} onSeeked={[Function: onSeeked]} onVolumeChanged={[Function: onVolumeChanged]} onLoadedMetadata={[Function: onLoadedMetadata]} preload=\\"metadata\\" style={{...}} title=\\"\\" volume={1} />
    </div>
  </div>
</nav>"
`;
